### PR TITLE
(#17) Added support for multiple file analysis; fixed filtering rule to be more readable.

### PR DESCRIPTION
**Pull request by silverCase**
File reading is implemented in a very strange manner via other dependencies.
This is a mere workaround.

### CODE REVIEW

**Comment by silverCase** [comment]
@mximp I don't understand the build failure:
```
Error:  /home/runner/work/polystat/polystat/src/main/java/org/polystat/Polystat.java:[158,40] cannot find symbol
  symbol:   variable Objects
  location: class org.polystat.Polystat
```
This location is corresponds to this code:
```
for (final Analysis analysis : Polystat.ALL) {
-------------------------------↑--------------
```

**Comment by mximp** [comment]
@silverCase can you please run `mvn -B verify -Pqulice` and make sure all warnings fixed.

File: src/main/java/org/polystat/Polystat.java [hunk 1]
```diff
@@ -153,38 +154,35 @@ public Integer call() throws Exception {
     private Iterable<Result> scan(final Path src, final Path tmp) {
         final Func<String, XML> xmir = new Program(src, tmp);
         final Collection<Result> errors = new ArrayList<>(Polystat.ALL.length);
+        final String extension = ".eo";
         for (final Analysis analysis : Polystat.ALL) {
             try {
-                errors.addAll(new ListOf<>(analysis.errors(xmir, "\\Phi.test")));
+                for (final String filename : Objects.requireNonNull(src.toFile().list())) {
```
Comment (mximp) [inline src/main/java/org/polystat/Polystat.java:160]: @silverCase There is not point in `requireNonNull`. If required check for empty folder explicitly.

File: src/main/java/org/polystat/Polystat.java [hunk 2]
```diff
@@ -153,38 +154,35 @@ public Integer call() throws Exception {
...
+                    if (filename.endsWith(extension)) {
+                        final String name = filename.split(extension)[0];
```
Comment (mximp) [inline src/main/java/org/polystat/Polystat.java:162]: @silverCase this var can be inlined

File: src/main/java/org/polystat/Polystat.java [hunk 3]
```diff
@@ -153,38 +154,35 @@ public Integer call() throws Exception {
...
+                        final String locator = String.format("\\Phi.%s", name);
```
Comment (mximp) [inline src/main/java/org/polystat/Polystat.java:163]: @silverCase same as above

File: src/main/java/org/polystat/Polystat.java [hunk 4]
```diff
@@ -153,38 +154,35 @@ public Integer call() throws Exception {
     private Iterable<Result> scan(final Path src, final Path tmp) {
         final Func<String, XML> xmir = new Program(src, tmp);
         final Collection<Result> errors = new ArrayList<>(Polystat.ALL.length);
+        final String extension = ".eo";
```
Comment (mximp) [inline src/main/java/org/polystat/Polystat.java:157]: @IngeniariusSoftware can be inlined

File: src/main/java/org/polystat/Polystat.java [hunk 5]
```diff
@@ -153,38 +154,35 @@ public Integer call() throws Exception {
...
-                errors.add(
-                    new Result.Failed(
-                        analysis.getClass(),
-                        ex,
-                        analysis.getClass().getName()
-                    )
-                );
+                final Class<? extends Analysis> clazz = analysis.getClass();
+                final String clazzname = clazz.getName();
+                final Result.Failed result = new Result.Failed(clazz, ex, clazzname);
```
Comment (mximp) [inline src/main/java/org/polystat/Polystat.java:172]: @silverCase all variables can be inlined

**Review by mximp — COMMENTED** [review]
@silverCase Can we create puzzle for writing tests for this

**Comment by silverCase** [comment]
@mximp if i make all those variables inlined - the readability will suffer drastically. Why would we want that?

**Comment by mximp** [comment]
> @mximp if i make all those variables inlined - the readability will suffer drastically. Why would we want that?

@silverCase these vars are used only once in each case. There is no need for dedicated variables. The code is more readable when it shows how the object (or function call) is constructed.

**Comment by mximp** [comment]
> > @silverCase Can we create puzzle for writing tests for this
> 
> I guess.? I don't know if 0pdd is up on this repository. But I know for sure that tests should be written. I can create a ticket for that.

I can see `.pdd` present. So PDD bot should be up.
If test creation is simple please go ahead. If it takes much time and code it's better to get another task for it and keep this PR short and targeted.

**Comment by silverCase** [comment]
@mximp, fixed everything, added tests - you can check.

File: src/test/java/org/polystat/PolystatITCase.java [hunk 6]
```diff
@@ -129,4 +151,23 @@ private static String exec(final String... cmds) throws Exception {
...
+     * @param name Name of a new file.
+     * @param resource Relative path of a resource to be copied.
+     * @throws Exception IOException - write operation failed, Exception - copying failed.
+    */
+    private static void writeFile(
+        final Path path,
+        final String name,
+        final String resource
+    ) throws Exception {
+        Files.write(
```
Comment (mximp) [inline src/test/java/org/polystat/PolystatITCase.java:168]: @silverCase We can use `TeeInput` & `OutputTo` from Cactoos library. This way we can get rid of static methods.

Comment (silverCase) [reply src/test/java/org/polystat/PolystatITCase.java:168]: Okay.

**Comment by silverCase** [comment]
@mximp please check.

**Review by mximp — APPROVED** [review]

**Comment by mximp** [comment]
@yegor256 please merge

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**